File: retrocast-client/Retrocast/Sources/Models/Role.swift
import Foundation

struct Role: Codable, Identifiable, Hashable, Sendable {
    let id: Snowflake
    let guildID: Snowflake
    var name: String
    var color: Int
    var permissions: Int64
    var position: Int
    var isDefault: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case guildID = "guild_id"
        case name
        case color
        case permissions
        case position
        case isDefault = "is_default"
    }
}

File: retrocast/internal/models/role.go
package models

type Role struct {
	ID          int64  `json:"id,string"`
	GuildID     int64  `json:"guild_id,string"`
	Name        string `json:"name"`
	Color       int    `json:"color"`
	Permissions int64  `json:"permissions,string"`
	Position    int    `json:"position"`
	IsDefault   bool   `json:"is_default"`
}
